### DISCUSSION TITLE
Alembic upgrade fails unexpectedly while using Google Cloud SQL proxy to connect

### DISCUSSION THREAD

**Discussion by digitalnomd** [sqlalchemy/alembic#1225] [Usage Questions]
Hey everyone :wave:, so I’m running into a bug with running the [python alembic package](https://alembic.sqlalchemy.org/en/latest/index.html) to perform a data migration on Google Cloud SQL. This requires first running a [Cloud SQL proxy](https://github.com/GoogleCloudPlatform/cloud-sql-proxy) in a subprocess to connect to my instance as a prerequisite and that process is being terminated before the the alembic operation completes.

From the logs below you can see the sequence of events:
```
04:10:28.52 [INFO] Starting CloudSQL Proxy for environment
04:10:28.81 [INFO] 2023/04/22 04:10:28 Listening on 127.0.0.1:39849 for myinstance
04:10:28.81 [INFO] 2023/04/22 04:10:28 Ready for new connections
04:10:28.81 [INFO] running Alembic upgrade to heads on port 39849...
04:11:23.34 [INFO] Terminating cloud sql proxy on port 39849
04:11:23.42 [ERROR] DBMigrationTask failed encountered unexpected exception: (pymysql.err.OperationalError) (2003, "Can't connect to MySQL server on '127.0.0.1' ([Errno 111] Connection refused)")
```

CloudSQL proxy is running on 127.0.0.1:39849 prior to beginning the Alembic process. I’m pretty sure that the error: `(2003, "Can't connect to MySQL server on '127.0.0.1' ([Errno 111] Connection refused)")` is because Cloud SQL proxy has terminated before Alembic completes. Behind the scenes, I'm using the `alembic.command.upgrade()` function to perform the upgrade.

Any ideas of how to debug this? Cloud SQL proxy is able to run fine during all other operations which makes me think that there’s an unsurfaced error that’s happening during the alembic upgrade process and those logs just aren’t being shown.

I’ve been stuck on this for a while and any help would be appreciated. Thanks!

For reference, I'm using alembic v1.6.5.

**Comment by CaselIT** [comment]
Hi,

That logging does not seem to came from alembic or sqlalchemy.
I would suggest increasing the verbosity of the package you are using and of sqlalchemy so that it logs the sql statement that are executed
